resource_leaves_geographic/models/resource_calendar.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from collections import defaultdict
+
+from odoo import models
+
+from odoo.addons.resource.models.resource import Intervals, make_aware
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    def _get_general_time_off_intervals_by_state(self, domain=None, any_calendar=False):
+        """Get a map of State to Intervals of leaves for the given domain."""
+        domain = domain or []
+        # Respect the any_calendar option
+        if not any_calendar:
+            domain += [("calendar_id", "in", self.ids + [False])]
+        # Ensure that the leaves are not related to a specific resource
+        domain += [
+            ("resource_id", "=", False),
+        ]
```

Comment:
issue: as a good rule of thumb, always put the safe domains first. Otherwise, you're open to [domain injection](https://www.youtube.com/live/1kt3Tf1HIQ8?feature=shared&t=988):

```suggestion
            domain = [("calendar_id", "in", self.ids + [False])] + domain
        # Ensure that the leaves are not related to a specific resource
        domain = [("resource_id", "=", False)] + domain
```